### PR TITLE
fix: add "queued" to the CensusSyncRunStatus enum

### LINKED ISSUE #25: CensusSyncRunStatus unable to handle certain statuses returned by Census API
Hey all,

we've been encountering issues using `run_census_sync`, as the Census API seems to return a sync status that is currently not supported by the Prefect Collection. This results in the flow failing, even though the sync status (`queued`) clearly indicates that there is no terminal issue at play here.

```
Encountered exception during execution:
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 833, in orchestrate_flow_run
    result = await flow_call.aresult()
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 292, in aresult
    return await asyncio.wrap_future(self.future)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 316, in _run_sync
    result = self.fn(*self.args, **self.kwargs)
  File "flows/activation/census_fbads_conversions.py", line 16, in census_fbads_conversions
    run_census_sync(census_sync)
  File "/usr/local/lib/python3.9/site-packages/prefect/flows.py", line 511, in __call__
    return enter_flow_run_engine_from_flow_call(
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 272, in enter_flow_run_engine_from_flow_call
    retval = from_sync.wait_for_call_in_loop_thread(
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/api.py", line 243, in wait_for_call_in_loop_thread
    return call.result()
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 283, in result
    return self.future.result(timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 169, in result
    return self.__get_result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 346, in _run_async
    result = await coro
  File "/usr/local/lib/python3.9/site-packages/prefect/client/utilities.py", line 51, in with_injected_client
    return await fn(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 717, in create_and_begin_subflow_run
    return await terminal_state.result(fetch=True)
  File "/usr/local/lib/python3.9/site-packages/prefect/states.py", line 91, in _get_state_result
    raise await get_state_exception(state)
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 833, in orchestrate_flow_run
    result = await flow_call.aresult()
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 292, in aresult
    return await asyncio.wrap_future(self.future)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 346, in _run_async
    result = await coro
  File "/usr/local/lib/python3.9/site-packages/prefect_census/flows.py", line 20, in run_census_sync
    sync_status = await task(census_sync_run.wait_for_completion.aio)(census_sync_run)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/api.py", line 182, in wait_for_call_in_loop_thread
    return call.result()
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 283, in result
    return self.future.result(timeout=timeout)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 169, in result
    return self.__get_result()
  File "/usr/local/lib/python3.9/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 346, in _run_async
    result = await coro
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1302, in get_task_call_return_value
    return await future._result()
  File "/usr/local/lib/python3.9/site-packages/prefect/futures.py", line 237, in _result
    return await final_state.result(raise_on_failure=raise_on_failure, fetch=True)
  File "/usr/local/lib/python3.9/site-packages/prefect/states.py", line 91, in _get_state_result
    raise await get_state_exception(state)
  File "/usr/local/lib/python3.9/site-packages/prefect/engine.py", line 1719, in orchestrate_task_run
    result = await call.aresult()
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 292, in aresult
    return await asyncio.wrap_future(self.future)
  File "/usr/local/lib/python3.9/site-packages/prefect/_internal/concurrency/calls.py", line 346, in _run_async
    result = await coro
  File "/usr/local/lib/python3.9/site-packages/prefect_census/syncs.py", line 375, in wait_for_completion
    CensusSyncRunStatus(self.status).name,
  File "/usr/local/lib/python3.9/enum.py", line 384, in __call__
    return cls.__new__(cls, value)
  File "/usr/local/lib/python3.9/enum.py", line 702, in __new__
    raise ve_exc
ValueError: 'queued' is not a valid CensusSyncRunStatus
```

As the Census API sometimes does and sometimes does not return the `queued` status, the flow is very inconsistent in whether it successfully executes or stops with a failure.

The actual sync in Census still seems to be triggered and executes just fine.

### PR DESCRIPTION
**Pull request by parkedwards**
Closes https://github.com/PrefectHQ/prefect-census/issues/25

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-census/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [ ] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-census/blob/main/CHANGELOG.md)

### CODE REVIEW

**Review by zzstoatzz — APPROVED** [review]

**Comment by sean-lynch** [comment]
+1 from the Census side, this is a valid state. Thanks for the fix @parkedwards !

**Comment by parkedwards** [comment]
yooo @sean-lynch no problemo, but i had one quick question 

in the Census docs, under `GET /sync_runs`, I'm not seeing the `queued` status documented, only `preparing`

https://docs.getcensus.com/basics/developers/api/sync-runs

two q's
1. is `preparing` ALSO a valid status we should add here?
2. or is `preparing` now `queued` on the Census side?

image

File: tests/conftest.py [hunk 1]
```diff
@@ -73,6 +73,10 @@ def mock_successful_sync_with_wait(respx_mock):
         headers={"Authorization": "Bearer my_api_key"},
     ).mock(
         side_effect=[
+            Response(
+                200,
+                json={"data": {"id": 5, "sync_run_id": 12345, "status": "queued"}},
+            ),
```
Comment (parkedwards) [inline tests/conftest.py:79]: throwing this fixture state here, as the polling _**should**_ still wait til a completion state is detected

**Comment by sean-lynch** [comment]
Ah, thanks for flagging @parkedwards.`queued` is the correct state and replaces `preparing` so you don't need to add the latter. I've updated the docs to reflect that as well.

**Pull request merged**